ESLint setup

vkarpov15 opened the pull request:
<!--  Thanks for sending a pull request!  Before submitting:

1. Read our CONTRIBUTING.md guide
2. Rebase your PR if it gets out of sync with main
-->

**What this PR does**:

Set up ESLint with TypeScript support. Disable a bunch of rules, leaving only semicolon and whitespace changes. Based on conversations, I believe @kathirsvn prefers 4 space indent, so we can enforce that with ESLint.

**Which issue(s) this PR fixes**:
Fixes #114

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

vkarpov15 committed:
add eslint setup

vkarpov15 committed:
run `npm run lint -- --fix` to fix indentation and semicolon usage

kathirsvn commented:
Thanks for this PR @vkarpov15. This was on the queue for a while and it looks great now. 

Can we also add a section in the DEVGUIDE.MD on how this should be used and when this should be run(i.e. before every commit), etc? Also, can we add this as a validation in a CI job to verify if all files in a PR are formatted before merging? (may be thats a separate item?)

vkarpov15 committed:
add lint section to DEVGUIDE.md re: code review comment

vkarpov15 committed:
add lint workflow

vkarpov15 committed:
add checkout step to lint workflow

vkarpov15 commented:
@kathirsvn I added a lint workflow and added entry to DEVGUIDE.md, what do you think?

vkarpov15 committed:
enable 2 more eslint rules: prefer-const and no-unused-vars

Add exception for args that start with `_` to no-unused-vars: ESLint does that by default in JS, but apparently typescript-eslint does not

kathirsvn commented:
> @kathirsvn I added a lint workflow and added entry to DEVGUIDE.md, what do you think?

LGTM
